### PR TITLE
Contribute Dash for R documentation to master

### PR DESCRIPTION
**Pull request by CanerIrfanoglu**
This PR introduces several improvements to the [dash for R docs](https://dashr.plot.ly). In addition to the previous commits, major additions in this PR include: 

- Dash for R DAQ documents are merged to dashr-squashed via https://github.com/plotly/dash-docs/pull/595

- Dash for R Canvas documents are merged to dashr-squashed via https://github.com/plotly/dash-docs/pull/587

All the chapters are revised by cross checking the [Dash Python Docs](https://dash.plot.ly/) and following fixes are made:

1. **Landing Page**

- Removed Python references from _Dash HTML Components_, _Dash Cytoscape_ and _Build Your Own Components_ parts.

- Under **Dash Tutorial**, title _Part 4. Callbacks With State_ changed to _Part 4. More About Callbacks_

2. **Dash Tutorial - Part 1. Installation**. 
- Installation instructions updated
```
# Before
library(devtools) # devtools: Tools to Make Developing R Packages Easier
install_github('plotly/dashR') # The core dash backend
install_github('plotly/dash-html-components') # HTML components
install_github('plotly/dash-core-components') # Supercharged components
```

```
# After
library(devtools) # devtools: Tools to Make Developing R Packages Easier
# installs dashHtmlComponents, dashCoreComponents, and dashTable
# and will update the component libraries when a new package is released
install_github('plotly/dashR', upgrade = TRUE)
```

3. **Dash Tutorial - Part 2. Dash Layout**.
- Uncommented all instances of `#app$run_server()` to `app$run_server()`
- Under **Summary** section reference links for _dashCoreComponents_ & _dashHtmlComponents_ docs updated to point R docs instead of Python

- Modified **Calling Help** section content to display help options identical to the rest of the docs
4. **Dash Tutorial - Part 3. Basic Callbacks**
- Added missing subtitles Dash App Layout & Chained Callbacks
- Changed all instances of `#app$run_server()` to `app$run_server()`
- Updated link text at the end _Part 4: State to Dash Tutorial_ --> _Part 4: More about callbacks_

5. **Dash Tutorial - Part 4. More About Callbacks**
- Uncommented all instances of `#app$run_server()` to `app$run_server()`

6. **Dash Tutorial - Part 5. Interactive Graphing and Crossfiltering**
- Titles are changed to `###` (htmlH3) equivalent in `Markdown`
- Parallel Coordinates link updated from pointing Python docs to R docs 

7. **Dash Tutorial - Part 6. Sharing Data Between Callbacks**
- Uncommented all instances of `#app$run_server()` to `app$run_server()`

8. **Component Libraries - Dash Core Components**
- Uncommented all instances of `#app$run_server()` to `app$run_server()` 
- Removed all instances of ``` (3 ticks) at the end of the examples
- Code snippet under **Location** was commented out and causing infinite loading when uncommented. Fixed by using `dccMarkdown`

```
# Before
#library(dashCoreComponents)
#location = dccLocation(id= "url", refresh= FALSE)
```

```
# After
dccMarkdown("```r
library(dashCoreComponents)
location = dccLocation(id = 'url', refresh = FALSE)
```",
className = “example-container”)
```
- Added _Back to the Table of Contents_ link at the end

9. **Component Libraries - Dash DataTable**
- Fixed the first example _predefinedColumns_ with https://github.com/plotly/dash-docs/pull/674/commits/da0ea9159228a758a74214d90c58276f277215e8. This example was causing the page to stuck at updating status and eventually crash
- Removed all instances of ``` (3 ticks) at the end of the examples

10. **Component Libraries - Dash Canvas**
- 🎨 4/6 Examples
-  Since Dash Canvas utils does not exist for R yet, omitted 5th Example

11. **Component Libraries - Dash Cytoscape**
- Removed all instances of ``` (3 ticks) at the end of the examples

12. **Component Libraries - Dash Bio**
- On index page, enabled displaying the code for examples when in the image mode
- On index page & examples modified code for reading data from local files to URLs for reproducibility
- Removed all instances of ``` (3 ticks) at the end of the examples
- Uncommented all instances of `#app$run_server()` to `app$run_server()` 

13. **Beyond the Basics - URL Routing and Multiple Apps**
- Added htmlHr() tag at the end of the page

14. **Production - Deployment**
- Added htmlHr() tag at the end of the page

Post-merge checklist:

The master branch is auto-deployed to `dash.plot.ly`.
Once you have merged your PR, wait 5-10 minutes and check dash.plot.ly 
to verify that your changes have been made.

- [x] I understand

If this PR documents a new feature of Dash:

- [ ] Comment on the original Dash issue with a link to the new docs.
- [ ] Reply to any community thread(s) asking for this feature.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,5 +19,9 @@ bundle.js
 .idea
 .env
 .mypy_cache/
-
+.Rhistory
+.RData
 ignore/indicators.csv
+Rplot001.jpg
```
Comment (alexcjohnson) [inline .gitignore]: What's this? And the empty file `dashr/chapters/dash-bio/Rplot001.jpg` that nonetheless shows up in this PR?

File: README.md [hunk 2]
```diff
@@ -10,10 +10,9 @@ To run an app locally:
 
 1. (optional) create and activate new virtualenv or conda env:
 
-```
-pip install virtualenv
-virtualenv venv
-source venv/bin/activate
+```r
+library(dash)
+heroku_app_template()
```
Comment (alexcjohnson) [inline README.md]: This doesn't belong in the main README, the original should be restored. Could be nice to create a similar doc for R (README-R could be that doc, but currently only has Heroku info - would need to have info about running locally and deploying to production) and _link_ to it from the main README. But that doesn't need to be done in this PR, for now we just need to restore the original README text.

File: python/Procfile.old [hunk 3]
```diff
@@ -0,0 +1 @@
+web: gunicorn run:server
```
Comment (rpkyle) [inline python/Procfile.old]: I think we can delete this one, but I'll confirm before doing so.

File: run.R [hunk 4]
```diff
@@ -0,0 +1,583 @@
...
+              headerStyle=list('color'='#0D76BF')
+            )
+          )
+        )
+      }
+    )
+  }
+)
+
+app$run_server(host = "0.0.0.0", port = Sys.getenv('PORT'))
```
Comment (rpkyle) [inline run.R]: I believe these items are now auto-populated thanks to the changes we've made to the buildpack and Dash for R; I ought to be able to :hocho: these settings.

File: python/Procfile.old (hunk 3 above)
Comment (rpkyle) [reply python/Procfile.old]: deleted in 79771a2b

File: run.R (hunk 4 above)
Comment (rpkyle) [reply run.R]: fixed in b05844a3

**Comment by rpkyle** [comment]
@alexcjohnson I believe this is ready for initial review. I'm not quite sure about the test failures, I'll need more time to look into them since they don't appear related (at first glance) to my edits on the R side -- but I'm able to successfully deploy the Python docs and R docs to separate Heroku instances.

The `search.js` now has conditional statements to handle R & Python in a single file; we're doing away with this and using our own custom search, so this isn't super important, but it seems to work on both sides.

File: app.R [hunk 5]
```diff
@@ -0,0 +1,10 @@
+library(dash)
+
+metas <- list(
+  list(name = 'description', 
+       content = 'Dash User Guide and Documentation. Dash is a framework for building analytical web apps in Python and R.')
```
Comment (chriddyp) [inline app.R]:
```suggestion
       content = 'Dash for R User Guide and Documentation. Dash is a framework for building analytical web apps in Python and R.')
```

File: app.R [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+)
+
+app <- Dash$new(assets_folder="dash_docs/assets", meta_tags = metas)
+
+app$title("Dash User Guide and Documentation - Dash by Plotly")
```
Comment (chriddyp) [inline app.R]:
```suggestion
app$title("Dash for R User Guide and Documentation - Dash by Plotly")
```

**Pull request merged**